internal/sbi/server.go

```diff
@@ -0,0 +1,199 @@
+		case "PATCH":
+			group.PATCH(route.Pattern, route.APIFunc)
+		case "DELETE":
+			group.DELETE(route.Pattern, route.APIFunc)
+		}
+	}
+}
+
+type pcf interface {
+	app.App
+	Processor() *processor.Processor
+	Consumer() *consumer.Consumer
+}
+
+type Server struct {
+	pcf
+
+	httpServer *http.Server
+	router     *gin.Engine
+}
+
+func NewServer(pcf pcf, tlsKeyLogPath string) (*Server, error) {
+	s := &Server{
+		pcf:    pcf,
+		router: logger_util.NewGinWithLogrus(logger.GinLog),
+	}
+
+	smPolicyRoutes := s.getSmPolicyRoutes()
+	smPolicyGroup := s.router.Group(factory.PcfSMpolicyCtlResUriPrefix)
+	applyRoutes(smPolicyGroup, smPolicyRoutes)
+
+	amPolicyRoutes := s.getAmPolicyRoutes()
+	amPolicyGroup := s.router.Group(factory.PcfAMpolicyCtlResUriPrefix)
+	amRouterAuthorizationCheck := util.NewRouterAuthorizationCheck(models.ServiceName_NPCF_AM_POLICY_CONTROL)
+	amPolicyGroup.Use(func(c *gin.Context) {
+		amRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(amPolicyGroup, amPolicyRoutes)
+
+	bdtPolicyRoutes := s.getBdtPolicyRoutes()
+	bdtPolicyGroup := s.router.Group(factory.PcfBdtPolicyCtlResUriPrefix)
+	bdtRouterAuthorizationCheck := util.NewRouterAuthorizationCheck(models.ServiceName_NPCF_BDTPOLICYCONTROL)
+	bdtPolicyGroup.Use(func(c *gin.Context) {
+		bdtRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(bdtPolicyGroup, bdtPolicyRoutes)
+
+	httpcallbackRoutes := s.getHttpCallBackRoutes()
+	httpcallbackGroup := s.router.Group(factory.PcfCallbackResUriPrefix)
+	applyRoutes(httpcallbackGroup, httpcallbackRoutes)
+
+	oamRoutes := s.getOamRoutes()
+	oamGroup := s.router.Group(factory.PcfOamResUriPrefix)
+	oamRouterAuthorizationCheck := util.NewRouterAuthorizationCheck(models.ServiceName_NPCF_OAM)
+	oamGroup.Use(func(c *gin.Context) {
+		oamRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(oamGroup, oamRoutes)
+
+	policyAuthorizationRoutes := s.getPolicyAuthorizationRoutes()
+	policyAuthorizationGroup := s.router.Group(factory.PcfPolicyAuthResUriPrefix)
+	policyAuthorizationRouterAuthorizationCheck := util.
+		NewRouterAuthorizationCheck(models.ServiceName_NPCF_POLICYAUTHORIZATION)
+	policyAuthorizationGroup.Use(func(c *gin.Context) {
+		policyAuthorizationRouterAuthorizationCheck.Check(c, pcf_context.GetSelf())
+	})
+	applyRoutes(policyAuthorizationGroup, policyAuthorizationRoutes)
+
+	uePolicyRoutes := s.getUePolicyRoutes()
+	uePolicyGroup := s.router.Group(factory.PcfUePolicyCtlResUriPrefix)
+	applyRoutes(uePolicyGroup, uePolicyRoutes)
+
+	s.router.Use(cors.New(cors.Config{
+		AllowMethods: []string{"GET", "POST", "OPTIONS", "PUT", "PATCH", "DELETE"},
+		AllowHeaders: []string{
+			"Origin", "Content-Length", "Content-Type", "User-Agent",
+			"Referrer", "Host", "Token", "X-Requested-With",
+		},
+		ExposeHeaders:    []string{"Content-Length"},
+		AllowCredentials: true,
+		AllowAllOrigins:  true,
+		MaxAge:           CorsConfigMaxAge,
+	}))
```

Comment:
these can be remove

Reply:
ok

Reply:
> @andy89923 or should it be moved to free5gc/pcf/internal/sbi/api_oam.go, where the cors headers are set?

Reply:
@HanHongChen Move cors related to OAM service.